hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/EvmTxProcessor.java

```diff
@@ -131,8 +133,13 @@ protected TransactionProcessingResult execute(
 
         // Enable tracing of contract actions if action sidecars are enabled and this is not a
         // static call
-        final HederaTracer hederaTracer =
-                new HederaTracer(!isStatic && isSideCarTypeEnabled(SidecarType.CONTRACT_ACTION));
+        final HederaTracer.EmitActionSidecars doEmitActionSidecars =
+                !isStatic && isSideCarTypeEnabled(SidecarType.CONTRACT_ACTION)
+                        ? EmitActionSidecars.ENABLED
+                        : EmitActionSidecars.DISABLED;
+        final HederaTracer.ValidateActionSidecars doValidateActionSidecars =
+                isSidecarValidationEnabled() ? ValidateActionSidecars.ENABLED : ValidateActionSidecars.DISABLED;
+        final HederaTracer hederaTracer = new HederaTracer(doEmitActionSidecars, doValidateActionSidecars);
```

Comment:
Much more readable with the `enum`s IMO 👍🏽

Reply:
Funny thing about sonar: Some things that (I consider) to be NOT code smells are marked as code smells, and some things that (I consider) TO BE code smells (like this was) are NOT marked that way.  If only I was in charge!

(IMO, even with IDE help to prompt with parameter names, boolean parameters are not that good, and if you have more than one on a method: that's to be avoided.  And Java/C#/C++ allow nice tightly scoped enums - C++ was the holdout but added `enum class` 10+ years ago ...)